feat: add search concept page (#105)
Basis-Theory/basistheory-integration-guides: collections/concepts/what-is-search.md

    width: 1200
    height: 630
---

# What is Search?

Encrypting data has always come with one costly tradeoff: it makes searching over your data very complicated. Basis Theory’s search feature aims to make that as simple and painless as possible, without any need to detokenize or decrypt your data.

## How It Works

When the Token is created the data is securely indexed in several data patterns using blind indexes. Combining the blind indexes with the existing metadata on the tokens allows you to search your entire Vault with a simple query. You can search over the following fields:

- `data`
- `type`
- `metadata.[key]`
- `created_at`

Currently, Basis Theory only supports searching the `data` for the `social_security_number` and `employer_id_number` [Token Types](https://docs.basistheory.com/#token-types). The data is indexed in several patterns, allowing for flexible searching. For instance, you can search for a social security number by the full number with or without dashes (`123-45-6789` and `123456789`), and only the last four (`6789`).

Each Token that matches the search query and is returned in the result set becomes an Active Token for that month.

## Permissions

> The search results are filtered based on the permissions associated with the Application Key. An Application requires at least one `read` [permission](https://docs.basistheory.com/#permissions-permission-types) to search for Tokens. In order to search the `data` field, the Application’s permissions must allow [unrestricted](https://docs.basistheory.com/#tokens-token-restriction-policies) access to plaintext Token data. Searching the `type`, `metadata`, and `created_at` fields only require a `read` permission.

Reviewer: ```suggestion
The search results are filtered based on the permissions associated with the Application. An Application requires at least one `read` [permission](https://docs.basistheory.com/#permissions-permission-types) to search for Tokens. Among the Tokens an Application has permission to read, the `type`, `metadata`, and `created_at` fields can be used to filter the search results. In order to filter by the `data` field, the Application’s permissions must allow [unrestricted](https://docs.basistheory.com/#tokens-token-restriction-policies) access to plaintext Token data.
```

## Query Syntax

Basis Theory uses a [Lucene-based query syntax](http://www.lucenetutorial.com/lucene-query-syntax.html) to power the search engine. Searching is case-insensitive, however only exact matches are currently supported. Wildcard searches are not available at this time. The search query is a string with one or more terms in the format `field:value`. For instance, if you want to find all social security numbers, you would query:

```
type:social_security_number
```

> To search Token `data` on Types that support it, you can for the indexed data patterns. To search for Tokens containing the data `123-45-6789`:

Reviewer: ```suggestion
To search Token `data` on Types that support it, you can search for the indexed data patterns. To search for Tokens containing the data `123-45-6789`, you would query:
```

```
data:123-45-6789
```

If you are searching over data that contains multiple words or spaces, you can wrap the search value in quotes:

```
data:"data containing multiple words"
```

> Searching to Tokens using metadata is supported as well. Metadata search terms use dot notation for key in the form of `metadata.key:value`. For example, to search for Tokens having the metadata `{ customer_id: "123456" }`, query for:

Reviewer: ```suggestion
Searching Tokens using metadata is supported as well. Metadata search terms use dot notation for key in the form of `metadata.key:value`. For example, to search for Tokens having the metadata `{ customer_id: "123456" }`, query for:
```
